src/Centreon/Application/Controller/AcknowledgementController.php

```diff
@@ -511,4 +516,101 @@ public function findAcknowledgements(RequestParametersInterface $requestParamete
             ]
         ])->setContext($context);
     }
+
+    /**
+     * Entry point to bulk acknowledge resources (hosts and services)
+     * @param Request $request
+     * @param EntityValidator $entityValidator
+     * @param SerializerInterface $serializer
+     * @return View
+     * @throws \Exception
+     */
+    public function massDisacknowledgeResources(
+        Request $request,
+        EntityValidator $entityValidator,
+        SerializerInterface $serializer
+    ): View {
+        $this->denyAccessUnlessGrantedForApiRealtime();
+
+        /**
+         * @var Contact $contact
+         */
+        $contact = $this->getUser();
+
+        if (!$contact->isAdmin() && !$contact->hasRole(Contact::ROLE_SERVICE_ACKNOWLEDGEMENT)) {
+            return $this->view(null, Response::HTTP_UNAUTHORIZED);
+        }
+
+        /**
+         * @var Resource[] $resources
+         */
+        $resources = $serializer->deserialize(
+            (string) $request->getContent(),
+            'array<' . Resource::class . '>',
+            'json'
+        );
+
+        $this->acknowledgementService->filterByContact($contact);
+
+        //validate input
+        $errorList = new ConstraintViolationList();
+        foreach ($resources as $resource) {
+            if ($resource->getType() === Resource::TYPE_SERVICE) {
+                $errorList->addAll($this->validateResource(
+                    $entityValidator,
+                    $resource,
+                    Resource::VALIDATION_GROUP_DISACK_SERVICE
+                ));
+            } elseif ($resource->getType() === Resource::TYPE_HOST) {
+                $errorList->addAll($this->validateResource(
+                    $entityValidator,
+                    $resource,
+                    Resource::VALIDATION_GROUP_DISACK_HOST
+                ));
+            } else {
+                throw new \RestBadRequestException('Incorrect resource type for disacknowledgement');
```

Comment:
maybe will be better to skip this resource instead to throw an exception because you have bulk action

Reply:
I do one loop to validate if all elements sent (a.k.a Request) are valid. If they are not the client is informed to fix their request. Once the data is validated, it goes to the service to apply action, and if something fails there - it is skipped as you suggested.